src/main/java/frc/robot/subsystems/Conveyor.java

```diff
@@ -0,0 +1,136 @@
+
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.utilities.Functions;
+import frc.robot.utilities.lists.Ports;
+
+/**
+ * Subsystem to control the conveyor of the robot.
+ */
+public class Conveyor extends SubsystemBase {
+
+    // motors
+    private final CANSparkMax front = new CANSparkMax(Ports.FRONT_CONVEYOR, MotorType.kBrushless);
+    private final CANSparkMax back = new CANSparkMax(Ports.BACK_CONVEYOR, MotorType.kBrushless);
+
+    // encoders
+    private final RelativeEncoder frontEncoder = front.getEncoder();
+    private final RelativeEncoder backEncoder = back.getEncoder();
+
+    public Conveyor() {}
+    
+    /**
+     * Sets the power of the front motor.
+     * @param power
+     */
+    public void setFrontMotorPower(double power) {
+        power = Functions.clampDouble(power, 1.0, -1.0);
+        front.set(power);
+    }
+
+    /**
+     * Sets the power of the back motor.
+     * @param power
+     */
+    public void setBackMotorPower(double power) {
+        power = Functions.clampDouble(power, 1.0, -1.0);
+        back.set(power);
+    }
+
+    /**
+     * Sets the power of both motors.
+     */
+    public void setMotorPower(double power) {
+        power = Functions.clampDouble(power, 1.0, -1.0);
+        front.set(power);
+        back.set(power);
+    }
+    
+    /**
+     * Gets the encoder position of the front motor (in rotations).
+     * @return position
+     */
+    public double getFrontEncoderPosition() {
+        return frontEncoder.getPosition();
+    }
+
+    /**
+     * Gets the encoder position of the back motor (in rotations).
+     * @return position
+     */
+    public double getBackEncoderPosition() {
+        return backEncoder.getPosition();
+    }
+
+    /**
+     * Manually sets the front encoder's position (in rotations).
+     * @param position
+     */
+    public void setFrontEncoder(double position) {
+        frontEncoder.setPosition(position);
+    }
+
+    /**
+     * Manually sets the back encoder's position (in rotations).
+     * @param position
+     */
+    public void setBackEncoder(double position) {
+        backEncoder.setPosition(position);
+    }
+
+    /**
+     * Gets the speed of the front motor (in RPM).
+     * @return speed
+     */
+    public double getFrontRPM() {
+        return frontEncoder.getVelocity();
+    }
+
+    /**
+     * Gets the speed of the back motor (in RPM).
+     * @return speed
+     */
+    public double getBackRPM() {
+        return backEncoder.getVelocity();
+    }
+
+    /**
+     * Resets the encoder values to 0.
+     */
+    public void zeroEncoders() {
+        setFrontEncoder(0);
+        setBackEncoder(0);
+    }
+
+    /**
+     * Sets how quickly the motors accelerate and decelerate.
+     * This is measured in the number of seconds it takes for the motor to ramp up to full speed.
+     * @param rate
+     */
+    public void setClosedRampRate(double rate) {
```

Comment:
Sorry, I gave you the wrong information at robotics today about the ramp rate stuff. When you set it it sets the ramp rate for when it is in a certain mode. For example, the closed ramp rate is for when it is doing PID and the open ramp rate is for when it is just getting set the power normally. This really should be setOpenRampRate
```suggestion
    public void setOpenRampRate(double rate) {
```